### ISSUE TITLE
Implement ComponentSelection Segmentation

### ISSUE THREAD

**Issue by jsbean** [jsbean/DNM#137]
For each `ViewerID`, create the default `ComponentSelection` for the `DurationInterval` that is equivalent to that of an entire piece.

Then for each user-initiated `ComponentSelection`, re-segment as such:

![componentselection](https://cloud.githubusercontent.com/assets/7040073/11598518/5a14d004-9a90-11e5-9385-3f050da7d406.png)

**Comment by jsbean** [comment]
First, implement 

```Swift
struct ComponentSelectionSequence: SequenceType {
    var componentSelections: [ComponentSelection] = []
    
    func addComponentSelection(componentSelection: ComponentSelection) {
        // deal with DurationInterval
    }
}
```

**Comment by jsbean** [comment]
Dealing with segmentation: 
- get segment that overlaps with `startDuration` of new segment
- get segment that overlaps with `stopDuration` of new segment

**Comment by jsbean** [comment]
![componentselection2](https://cloud.githubusercontent.com/assets/7040073/11600155/dc6f12dc-9a98-11e5-89b7-ed601f3224e4.png)

**Comment by jsbean** [comment]
`durationInterval` for `stopOverlapCopy` is instead `new.stopDuration --> stopOverlap.stopDuration`

**Comment by jsbean** [comment]
A good start at this is in 88c7acce5081e539f0b2be92bc516b3fa444aa6f. Still some refining and testing needed.

**Comment by jsbean** [comment]
Instead of replacing the selections that are overlapped with the new one, iterate through each selection overlapped and inject the keys that have changed.

Start with `secondPart` of the `startSelection` bisection, then overlapped selections (if any), then `firstPart` of the `stopSelection` bisection.

**Comment by jsbean** [comment]
For the UI, imagine the scenario where a new selection overlaps several.

Create a tableview for each `PerformerID`. For each `componentType`, there are 4 ui states:
- Show
- Hide
- Mute (to be shown, but overridden by the Performer being hidden)
- Multiple values (when there are multiple values for the key)

**Comment by jsbean** [comment]
Closed with 2f6a922572376e88c57c2321069fa7ccfb466b00
